.circleci/test.sh

```diff
@@ -6,7 +6,7 @@ set +o pipefail
 
 ROOT=$(dirname $0)/..
 EXIT_STATE=0
-MAX_AUTO_RETRY=5
```

Comment:
Does that mean with these fixes none of the tests are flaky anymore? If so that's fantastic! But I wonder whether that really applies to all uses of `retry` - it's used in `jasmine2`, `jasmine3`, and `image2` and I see lots of changes here related to jasmine tests but none related to image tests, perhaps it's safer to set `MAX_AUTO_RETRY` separately for each case?

Also I see at least one commit https://github.com/plotly/plotly.js/pull/5340/commits/5769cde5b22471aa0d83d62feae2c378b2ee9358 about moving tests from `flaky` to `noCI` - I don't think that's a good idea, long-term we should be working to get rid of `noCI` unless it really *can't* run on CI, but if it's just flaky there we should run it with retry as we have been. So maybe really what we should be doing here is re-evaluating which tests need the `flaky` tag? If in this PR we can remove that tag from many of the tests that's a win even if we can't remove it everywhere.

Reply:
Yes. They are less flaky right now.
Good call. I was able to removed a number of flaky tags in https://github.com/plotly/plotly.js/pull/5340/commits/9753fb95e6ecfae2172895be289576e2cb3c0ff2
and two noCI flags in https://github.com/plotly/plotly.js/pull/5340/commits/bbe25efe84d6e91b7cf699e50c1cdc6688120435 and https://github.com/plotly/plotly.js/pull/5340/commits/70ec83a4f8f678dfcb61be19cba0c977d784dc9e.

Concerning `image2` we could make a separate `MAX_AUTO_RETRY` if/when we noticed any problem; but I am not too worried of that.

Reply:
What about the four newly added `@noCI` tags in [`gl3d_hover_click_test`](https://github.com/plotly/plotly.js/pull/5340/files#diff-d074d40f68a594d0e55cccf93bc4897c74cb607a87791ef0c991acd19222c010) and the two new `@noCI` at the start of [`parcoords_test`](https://github.com/plotly/plotly.js/pull/5340/files#diff-98e1462b9d29ba2c756f1371541f23d14a9cdc1b0d3986b6d9ad1c4cf81fd174)?

Reply:
> What about the four newly added `@noCI` tags in [`gl3d_hover_click_test`](https://github.com/plotly/plotly.js/pull/5340/files#diff-d074d40f68a594d0e55cccf93bc4897c74cb607a87791ef0c991acd19222c010) and the two new `@noCI` at the start of [`parcoords_test`](https://github.com/plotly/plotly.js/pull/5340/files#diff-98e1462b9d29ba2c756f1371541f23d14a9cdc1b0d3986b6d9ad1c4cf81fd174)?

Those GL tests are quite flaky and I noticed them to often fail on other branches.
Since we cannot add a flaky flag for GL tests (at least at the moment), I decided to add those `@noCI` tags.
At one point we should possibly rewrite/adjust those tests to be more stable.

Reply:
OK, thanks for the details. This is a net reduction in `@noCI` and `@flaky` regardless 🎉

Reply:
FYI - I was able to drop newly added `@noCI` flags from `parcoords` tests in https://github.com/plotly/plotly.js/pull/5340/commits/ce51c692ea085d4a57e2196c98df0955c88a4a8f.